docs/ELIGIBILITY_FLOW.md

```diff
@@ -0,0 +1,134 @@
+# Eligibility Flow
+
+The eligibility flow is the sequence of events used to determine if an upkeep is
+considered eligible to perform. This involves some input on a regular interval,
+a series of pre-processing, checking via RPC calls, collecting data from HTTP
+calls, and finally some type of post-processing. The output is expected to be
+only upkeeps eligible to be performed on chain.
+
+Required data includes a collection of active, configured upkeeps with an
+unknown eligibility status.
+
+[Generic Sequence Diagram](./diagrams/generic_ticker_sequence.md)
+
+## Basic Value Types
+
+To maintain chain agnostic behavior, the following data types should be
+abstracted from the plugin as much as possible. That is, the less detail the
+plugin needs from each of the following value types, the better.
+
+### ConfiguredUpkeep
+
+A configured upkeep represents all information that identifies a single upkeep
+as registered by the user including upkeep id, trigger configuration, 
+check data, etc.
+
+```golang
+type UpkeepIdentifier []byte
+
+type ConfiguredUpkeep struct {
+    // ID uniquely identifies the upkeep
+    ID UpkeepIdentifier
+    // Type is the event type required to initiate the upkeep
+    Type int
+    // Config is configuration data specific to the type
+    Config interface{}
+}
+```
+
+### UpkeepPayload
+
+An upkeep payload represents all information the check pipeline requires to
+check the eligibility of an upkeep and might include upkeep id, block number,
+check data, etc.
+
+This can potentially be a combination of both `ConfiguredUpkeep` and `T`:
+
+```golang
+type UpkeepPayload struct {
+    // Upkeep is all information that identifies the upkeep
+    Upkeep ConfiguredUpkeep
+    // Tick is the event that triggered the upkeep to be checked
+    Tick interface{}
```

Comment:
Would be nice to use actual type here (generic Tick) rather than interface

Reply:
Same here. The payload is being passed to the registry. Not sure how to do generics when there are multiple typed values.